### ISSUE TITLE
Color Inconsistency (Grab has bugfix)

### ISSUE THREAD

**Issue by wipfli** [maplibre/maplibre-gl-native#568]
## Summary

Ensure same color rendering results between style, layer and annotation.

## Grab has bugfix

The original author of this bug report is Grab. They have a bug fix and would like to upstream it to MapLibre GL Native.

**Comment by cbeddow** [comment]
Following, it could be what I have here: https://twitter.com/cbed32/status/1596533172350484480?s=20&t=globat9csUMQbP21x7fI6Q

**Comment by alanchenboy** [comment]
Sorry for the false report, This issue happened when I tried to make the color consistent with the style and programmable. In Mapbox when you create a color with an alpha value(not 1), then the style color is the normal color, for example, the fill color :
```
        {
            "id": "a-fill-layer",
            "type": "fill",
            ....
            "paint": {"fill-color": "rgba(1.0,0.0,0.0,0.5)"}
        },
```
But when you create the layer manually the color must be the pre-multiplied color: Color.premultiply = (Color.rgb * color.a, color.a);

Your code in Objective C should be like this(@cbeddow your issue may be related to this):

```
    MGLFillStyleLayer *fillLayer = [[MGLFillStyleLayer alloc] initWithIdentifier:@"test" source:source];
    fillLayer.fillColor = [NSExpression expressionForConstantValue:[UIColor colorWithRed:1.0*0.4 green:0.0*0.5 blue:0.0*0.5 alpha:0.5];

```

I deprecated this using method in GrabMap, the fill color needn't use pre-multiply color, and this issue happened when I solve this I miss the uniform color value interpolating. So if Maplibre still wants to use pre-multiply color in the core, shouldn't care about this, it's no issue with premultiply color.
